python/adelphi/adelphi/nb.py

```diff
@@ -0,0 +1,127 @@
+import yaml
+
+from itertools import chain
+
+from adelphi.exceptions import TooManyKeyspacesException, TooManyTablesException
+from adelphi.export import BaseExporter
+
+SEQS={}
+SEQS["text"] = "Mod(1000000000); ToString() -> String"
+DISTS={}
+DISTS["text"] = "Hash(); Mod(1000000000); ToString() -> String"
+
+# A collection of functionally static functions
+def dist_binding_name(col):
+    return "{}_dist".format(col.name)
+
+
+def seq_binding_name(col):
+    return "{}_seq".format(col.name)
+
+
+def build_bindings(table):
+    rv = {}
+    for (col_name, col) in table.columns.items():
+        if col.cql_type not in DISTS.keys():
+            raise UnsupportedColumnTypeException("No nosqlbench distribution configured for type {}".format(col.cql_type))
+        rv[dist_binding_name(col)] = DISTS[col.cql_type]
+
+        # If we have a primary key we'll also need a seq for insert statements
+        if col_name in [c.name for c in table.primary_key]:
+            if col.cql_type not in SEQS.keys():
+                raise UnsupportedColumnTypeException("No nosqlbench sequence configured for type {}".format(col.cql_type))
+            rv[seq_binding_name(col)] = SEQS[col.cql_type]
+
+    return rv
+
+
+def build_select_statements(keyspace, table):
+    key_bindings = " and ".join(["{} = {}".format(key.name, "{" + dist_binding_name(key) + "}") for key in table.primary_key])
+    return "select * from  {}.{} where {}".format(keyspace.name, table.name, key_bindings)
+
+
+def build_insert_statements(keyspace, table):
+    cols = table.columns.values()
+    primary_keys = set(table.primary_key)
+    col_names = ",".join([col.name for col in cols])
+    def binding_name(col):
+        return seq_binding_name(c) if col in primary_keys else dist_binding_name(c)
+    col_bindings = ",".join(["{" + binding_name(c) + "}" for c in cols])
+    return "insert into {}.{} ({}) values ({})".format(keyspace.name, table.name, col_names, col_bindings)
+
+
+class UnsupportedColumnTypeException(Exception):
+    """Exception indicinating that we can't create nosqlbench structures for a column type"""
+    def __init__(self, msg):
+        self.message = msg
```

Comment:
adelphi.exceptions defines exceptions common to all commands... it's also entirely fine for a command to define their own custom exceptions as necessary.